services/tenant-ui/frontend/src/components/verifier/VerifiedPresentationData.vue

```diff
@@ -0,0 +1,155 @@
+
+  //normalize attribute_groups for table
+  for (const [k, v] of Object.entries(requested_attribute_groups())) {
+    // @ts-expect-error types have not been defined for this object
+    v.names.forEach((name: string) => {
+      result.push({
+        name: name,
+        val: pres.acapy.presentation_exchange.presentation.requested_proof
+          .revealed_attr_groups[k].values[name].raw,
+        attr_type: 'requested_attribute_group',
+        referent: k,
+        checkmark: true,
+        // @ts-expect-error types have not been defined for this object
+        restrictions: v.restrictions,
+      });
+    });
+  }
+  //normalize single_attributes for table
+  for (const [k, v] of Object.entries(requested_single_attributes())) {
+    result.push({
+      // @ts-expect-error types have not been defined for this object
+      name: v.name,
+      val: pres.acapy.presentation_exchange.presentation.requested_proof
+        .revealed_attrs[k].raw,
+      attr_type: 'requested_single_attribute',
+      referent: k,
+      checkmark: true,
+      // @ts-expect-error types have not been defined for this object
+      restrictions: v.restrictions,
+    });
+  }
+  //normalize self_attested_attributes for table
+  for (const [k, v] of Object.entries(requested_self_attested_attributes())) {
+    result.push({
+      // @ts-expect-error types have not been defined for this object
+      name: v.name,
+      val: pres.acapy.presentation_exchange.presentation.requested_proof
+        .self_attested_attrs[k],
+      attr_type: 'self_attested_attribute',
+      referent: k,
+      checkmark: false,
+      restrictions: { info: 'No Restrictions' },
+    });
+  }
+
+  // normalize requested_predicates for table
+  // ag_rows = requested_attribute_groups().map(ua => console.log(ua)
+  return result;
+};
+// four different payload locations for the provided claims based on these filters.
+const requested_attribute_groups = (): any => {
+  return Object.fromEntries(
+    Object.entries(
+      props.presentation.acapy.presentation_exchange.presentation_request
+        .requested_attributes
+    ).filter(([key, ra]: [string, any]): any => {
+      return 'names' in ra && 'restrictions' in ra;
+    })
+  );
+};
+
+const requested_single_attributes = (): any => {
+  return Object.fromEntries(
+    Object.entries(
+      props.presentation.acapy.presentation_exchange.presentation_request
+        .requested_attributes
+    ).filter(([key, ra]: [string, any]): any => {
+      return 'name' in ra && 'restrictions' in ra;
+    })
+  );
+};
+
+const requested_self_attested_attributes = (): any => {
+  return Object.fromEntries(
+    Object.entries(
+      props.presentation.acapy.presentation_exchange.presentation_request
+        .requested_attributes
+    ).filter(([key, ra]) => {
+      // @ts-expect-error types have not been defined for this object
+      return 'name' in ra && !('restrictions' in ra);
+    })
+  );
+};
```

Comment:
There's probably a bunch of utility-function-ing that can be done here to reduce it, but I'd be inclined to leave it until we done have as much in-flight or proposed changes still to Presentation UX.

Reply:
Yea, i know the JS here is rudimentary, but i'm also confident that any developer can understand this code. Let me know if you want me to make any changes here now.